flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -4,247 +4,278 @@ import UIKit
-              sdk["packages"] = sdkPackages.append(contentsOf: packages)
-            } else {
-              sdk["packages"] = packages
+            if let sentryOptions = self.sentryOptions {
+                infos["package"] = ["version": sentryOptions.sdkInfo.version, "sdk_name": "cocoapods:sentry-cocoa"]
             }
-          }
 
-          if let integrations = arguments["integrations"] as? [String] {
-            if var sdkIntegrations = sdk["integrations"] as? [String] {
-              sdk["integrations"] = sdkIntegrations.append(contentsOf: integrations)
-            } else {
-              sdk["integrations"] = integrations
+            result(infos)
+        }
+    }
+
+    private func initNativeSdk(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        guard let arguments = call.arguments as? [String: Any], !arguments.isEmpty else {
+            print("Arguments is null or empty")
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil))
+            return
+        }
+
+        SentrySDK.start { options in
+            self.updateOptions(arguments: arguments, options: options)
+
+            self.sentryOptions = options
+
+            // note : for now, in sentry-cocoa, beforeSend is not called before captureEnvelope
+            options.beforeSend = { event in
+                self.setEventOriginTag(event: event)
+
+                if var sdk = event.sdk, self.isValidSdk(sdk: sdk) {
+                    if let packages = arguments["packages"] as? [String] {
+                        if var sdkPackages = sdk["packages"] as? [String] {
+                            sdk["packages"] = sdkPackages.append(contentsOf: packages)
+                        } else {
+                            sdk["packages"] = packages
+                        }
+                    }
+
+                    if let integrations = arguments["integrations"] as? [String] {
+                        if var sdkIntegrations = sdk["integrations"] as? [String] {
+                            sdk["integrations"] = sdkIntegrations.append(contentsOf: integrations)
+                        } else {
+                            sdk["integrations"] = integrations
+                        }
+                    }
+                    event.sdk = sdk
+                }
+
+                return event
             }
-          }
-          event.sdk = sdk
         }
 
-        return event
-      }
+        if startSession && sentryOptions?.enableAutoSessionTracking == true {
+            // only this is not enough, we need to mimics SentrySessionTracker.didBecomeActive
+            // how to call that from here?
+            // it'd also be nice to know if theres already a running session, so we dont do twice, lets say that
+            // not always the SDK is init. too late.
+            SentrySDK.currentHub().startSession()
+        }
+
+        result("")
     }
```

Comment:
after discussing on a call, it'd be nice that Cocoa has a custom notification that mimics the `didBecomeActive` behaviour, so we don't need to expose anything and could also be used by the RN SDK.
ideally this notification would accept a timestamp and use that timestamp to start the session.

Reply:
https://github.com/getsentry/sentry-cocoa/issues/912